databases/gorm/posts.go

```diff
@@ -1,33 +1,65 @@
 package database
 
-import "github.com/arshamalh/blogo/models"
+import (
+	"log"
+
+	"github.com/arshamalh/blogo/models"
+)
 
 func (gdb *gormdb) CreatePost(post *models.Post) (uint, error) {
-	err := gdb.db.Create(&post).Error
+	err := gdb.db.Create(post).Error
+
+	if err == nil {
+		log.Printf("Post created with ID: %d", post.ID)
+	} else {
+		log.Printf("Failed to create post: %v", err)
+	}
+
 	return post.ID, err
 }
 
 func (gdb *gormdb) DeletePost(id uint) error {
-	return gdb.db.Delete(&models.Post{}, "id = ?", id).Error
+	err := gdb.db.Delete(&models.Post{}, id).Error
+
+	if err == nil {
+		log.Printf("Post with ID %d deleted", id)
+	} else {
+		log.Printf("Failed to delete post with ID %d: %v", id, err)
+	}
+
+	return err
 }
 
 func (gdb *gormdb) UpdatePost(post *models.Post) error {
-	return gdb.db.Updates(&post).Error
+	err := gdb.db.Save(post).Error
+
+	if err == nil {
+		log.Printf("Post with ID %d updated", post.ID)
+	} else {
+		log.Printf("Failed to update post with ID %d: %v", post.ID, err)
+	}
+
+	return err
 }
 
 func (gdb *gormdb) GetPost(id uint) (models.Post, error) {
-	// FIXME: This is not the best way to do this
 	var post models.Post
-	err := gdb.db.
-		Preload("Author").
-		Preload("Category").
-		Preload("Comment").
-		First(&post, id).Error
+	err := gdb.db.Preload("Author").Preload("Category").Preload("Comment").First(&post, id).Error
```

Comment:
Long chains are more readable when they are in multiple lines, I used to stack them in one line, but now I realize lines are ours! so use them for cleaning the project,

But, this part of the code is a mess in some other aspects (db joins) that we will fix it later.

Reply:
ok